### PR TITLE
updated schema, changed usage, repeated code to quickly check if it w…

### PR DESCRIPTION
**Pull request by ahtoms**
There are some TODOs scattered within there but these are the modifications required with the latest changes.

### CODE REVIEW

**Review by mmcgr — APPROVED** [review]

File: src/DwarfMap.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
+
+#ifndef SRC_DL_DWARFMAP_H_
+#define SRC_DL_DWARFMAP_H_
+
+#include <cstdlib>
+#include <map>
+#include <stdint.h>
+#include <libdwarf/libdwarf.h>
+#include <dwarf.h>
```
Comment (azreika) [inline src/DwarfMap.hpp:9]: This seems to only be compiling for me if I change this line to `#include <libdwarf/dwarf.h>`. Is this an error with the includes, or something else?

File: src/disasm_main.cpp [hunk 2]
```diff
@@ -38,6 +38,7 @@
 #include "passes/FunctionInferencePass.h"
 #include "passes/NoReturnPass.h"
 #include "passes/SccPass.h"
+#include "DwarfMap.hpp"
```
Comment (azreika) [inline src/disasm_main.cpp:41]: The other CPP header files have a `.h` extension, should change for consistency.

File: src/GtirbZeroBuilder.cpp [hunk 3]
```diff
@@ -114,13 +114,16 @@ gtirb::from_iterator gtirb::auxdata_traits<ElfSymbolInfo>::fromBytes(ElfSymbolIn
 void buildSections(gtirb::Module &module, std::shared_ptr<BinaryReader> binary,
                    gtirb::Context &context)
 {
+    std::map<gtirb::UUID, SectionProperties> dwarfSections;
+    std::map<gtirb::UUID, SectionProperties> allSections;
```
Comment (azreika) [inline src/GtirbZeroBuilder.cpp:118]: This doesn't seem to be updated - should this be a union of `dwarfSections` and `sectionProperties`?

File: src/AuxDataSchema.h [hunk 4]
```diff
@@ -84,13 +84,33 @@ namespace gtirb
...
+        };
+
         /// \brief Auxiliary data covering ELF section properties.
         struct ElfSectionProperties
         {
             static constexpr const char* Name = "elfSectionProperties";
             typedef std::map<gtirb::UUID, std::tuple<uint64_t, uint64_t>> Type;
         };
 
+        struct AllElfSectionProperties
```
Comment (azreika) [inline src/AuxDataSchema.h:100]: Why are three `.*ElfSectionProperties` auxiliary tables needed? In particular, if `AllElfSectionProperties` is just a union of the other two, what's the benefit of keeping it?

For the ones that stay, we should add `/// \brief` style comments at some point to clarify the distinction.

**Review by azreika — COMMENTED** [review]
This has already been merged in, but added a few comments for future PR fixes.

Also, it does not currently build on my machine due to the header issue mentioned above. If fixed, the program also crashes with a segfault on examples due to an assertion error:

```
void gtirb::AuxDataContainer::addAuxData(typename Schema::Type&&) [with Schema = gtirb::schema::AllElfSectionProperties; typename Schema::Type = std::map<boost::uuids::uuid, std::tuple<long unsigned int, long unsigned int> >]: Assertion `TypeEntry != TypeMap.Map.end() && TypeEntry->second->getApiTypeId() == AuxDataImpl<Schema>::staticGetApiTypeId() && "Attempting to add AuxData with unregistered or incorrect type."' failed.
```

**Pull request merged**